packages/samples/headless-react/src/components/cart-recommendations-list/cart-recommendations-list.class.tsx

```diff
@@ -0,0 +1,135 @@
+import {Component} from 'react';
+import {
+  buildCartRecommendationsList,
+  CartRecommendationsList as HeadlessCartRecommendationsList,
+  CartRecommendationsListState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {productRecommendationEngine} from '../../engine';
+
+export interface CartRecommendationsListProps {
+  skus: string[];
+  maxNumberOfRecommendations?: number;
+}
+
+export class CartRecommendationsList extends Component<
+  CartRecommendationsListProps
+> {
+  private controller: HeadlessCartRecommendationsList;
+  public state: CartRecommendationsListState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: CartRecommendationsListProps) {
+    super(props);
+
+    this.controller = buildCartRecommendationsList(
+      productRecommendationEngine,
+      {
+        options: {
+          skus: this.props.skus,
+          maxNumberOfRecommendations: this.props.maxNumberOfRecommendations,
+        },
+      }
+    );
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
```

Comment:
@francoislg This `render` is copy and pasted across two different samples, that's normal since we want to show how to use the component as a class or functional component.